### PR TITLE
Added fixer dfmt for the D programming language

### PR DESCRIPTION
**Pull request by bkhl**
Support for dfmt (https://github.com/dlang-community/dfmt)

### CODE REVIEW

File: doc/ale-d.txt [hunk 1]
```diff
@@ -13,6 +13,27 @@ g:ale_d_dls_executable                                 *g:ale_d_dls_executable*
...
+dfmt                                                               *ale-d-dfmt*
+
+g:ale_d_dfmt_executable                               *g:ale_d_dfmt_executable*
+                                                      *b:ale_d_dfmt_executable*
+  Type: |String|
+  Default: `dfmt`
+
+See |ale-integrations-local-executables|
+
+g:ale_d_dfmt_options                                     *g:ale_d_dfmt_options*
```
Comment (w0rp) [inline doc/ale-d.txt]: I recommend putting two blank lines between each option.

Comment (bkhl) [reply doc/ale-d.txt]: Done in https://github.com/dense-analysis/ale/pull/2712/commits/092d553e671b3d39f5973c2ae8f84ccc878f12dc

**Comment by bkhl** [comment]
Do I need to do anything else here? Should I recreate the branch to minimize the size of the merge?

**Comment by w0rp** [comment]
Someone else also submitted a pull request for dfmt, and it looks like I already merged that one. Thanks!

It's good having support for `dfmt`. I used to be an avid D user myself.

**Comment by bkhl** [comment]
I see, though having a look at https://github.com/dense-analysis/ale/blob/2e323b529d8a4938f2077f31b937bc47c6c180bc/autoload/ale/fixers/dfmt.vim I think mine is better, as there is no need for a temporary file for the dfmt fixer. Should I adjust this PR or create a new one?

**Comment by w0rp** [comment]
Create a new PR and I'll have a look when I can.

**Pull request closed without merging**